### DISCUSSION TITLE
Multiple pimd instances

### DISCUSSION THREAD

**Discussion by gMeCode** [troglobit/pimd#205] [General]
Hello,

I am using pimd v2.3.2

I am relatively new to pimd and multicast in general. I was able to setup and configure a system
with a point to point VPN (VPN Remote 1 and VPN Server below). This all works well. I can request and 
stream multicast across the link in both directions. 
NETA -- [ VPN Remote 1 -- 10.1.1.2/24 ]  --------------------- [VPN Server 10.1.1.1/24] -- NETB

When I add a second remove as shown below I am having some issues. 
NETA -- [ VPN Remote 1 -- 10.1.1.2/24 ]  --------------------- [VPN Server 10.1.1.1/24] -- NETB
NETC -- [ VPN Remote 2 -- 10.1.1.3/24 ] --------------------- 

Notes that on the server this is a single interface. 
The server shows "PIM" as the FLAGS for both remotes using pimd -r and shows the correct
IP Address for the "Neighbors"

But in the route tables I am only seeing the multicast addresses for one of the remotes (these are the 
addresses that I configured on the remotes using the group-prefix. 

I assume that the issue is that the pimd instance running on the server can not communicate with 
two instances of pimd running the remotes? From there I started to look at the idea of running two instances
of pimd on the server. one instance communicating with each remote and I noticed that as of v2.3.0 you can run multiple instances of pimd

"Support for multiple multicast routing tables, and running multiple pimd instances"

I also saw the note regarding the "-I Ident" argument but when I use that pimd throws an error. 

I am happy to supply config files or anything else that would help. Initially I was looking to see if I was on track and what I am missing.

**Comment by troglobit** [comment]
It may be that 10.1.1.2 and 10.1.1.3 maps to the same physical interface. This doesn't work if you want to run multiple instances since the virtual interfaces (VIFs) used by the kernel, can only map to a unique physical interface which cannot be shared between instances. 

*Maybe* you can solve this by creating a macvlan interface or something on top of this shared interface, but this is very much untested. However, why can't you use the same pimd instance for both VPN tunnel interfaces?  The destination (or source?) network seem to be shared anyway.

**Comment by gMeCode** [comment]
Thank you for the feedback! I found the root of the issue. The way that the VPN is setup I can not have a route for a Network to two different remote devices. So on the server I created a second interface. So now it looks more like this

NETA -- [ VPN Remote 1 -- 10.1.1.2/24 ] --------------------- [VPN Server iface B -- 10.1.1.1/24] -- NETB
NETC -- [ VPN Remote 2 -- 10.2.1.3/24 ] ---------------------[VPN Server iface D -- 10.2.1.1/24] 

I thought that previously I was having issues with pimd communicating with peers on different interfaces, which is why I was trying to use a single interface, but that was an issue in my configuration. The discussion "Multicast via OpenVPN tunnel #77" help a lot. 

Thank you!
